discovery_engine_core/core/src/engine.rs

```diff
@@ -517,54 +518,95 @@ impl Engine {
         &mut self,
         history: Option<&[HistoricDocument]>,
         sources: &[WeightedSource],
-        reacted: &UserReacted,
-    ) -> Result<(), Error> {
+        reacted: UserReacted,
+    ) -> Result<Document, Error> {
+        let reaction = reacted.reaction;
+        cfg_if! {
+            if #[cfg(feature="storage")] {
+                let document: Document = self.storage
+                    .feedback()
+                    .update_user_reaction(reacted.id, reacted.reaction).await?
+                    .into();
+            } else {
+                use chrono::Utc;
+                use url::Url;
+
+                use crate::document::NewsResource;
+
+                let document = Document {
```

Comment:
We fake a document to have a uniform interface in the remaining code, IMHO in the future (post migration) the code should directly use `ApiDocumentView` instead of `document::Document`.